fast64_internal/sm64/sm64_geolayout_classes.py

```diff
@@ -17,8 +17,10 @@
 }
 
 def getDrawLayerName(drawLayer):
-	if drawLayer in drawLayerNames:
-		return drawLayerNames[drawLayer]
+	# Cast draw layer to int so it can be mapped to a name
+	layer = int(drawLayer) if isinstance(drawLayer, str) and drawLayer.isnumeric() else drawLayer
```

Comment:
haha those tabs sizes are HUGE! and i _think_ that isnumeric is what i'm looking for https://docs.python.org/3/library/stdtypes.html#str.isnumeric

your method makes sense to me, though i think the try/catch is a little bulky. what i was trying to do here is check if drawLayer is a string that only contains numbers. the other possible values here are C macros like `"LAYER_TRANSPARENT"`. I'm not exactly sure how else a `ValueError` could happen here but if a ValueError popped up with this specific condition i think that i would want it to throw an error because it would be some wildly unexpected value. i'm open to changing it though

Reply:
`isnumeric` allows many more codepoints than just 0-9 characters
> Numeric characters include digit characters, and all characters that have the Unicode numeric value property, e.g. U+2155, VULGAR FRACTION ONE FIFTH.
for example with U+2155
```
>>> int("\u2155")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: '⅕'
```

it's quite common to use `except ValueError` to detect if something can't be converted
maybe you would prefer something like `all(c >= "0" and c <= "9" for c in drawLayer)`

Reply:
ahhh yeah, good point. i'll go with the ValueError check then b/c its probably a bit cheaper